### DISCUSSION TITLE
Bug or user error? Local development with email auth has wrong URL in template email

### DISCUSSION THREAD

**Discussion by tomelliot** [supabase/supabase#15936] [Questions] [answered]
Hi there,

I'm using the containerised local development set up and just configured email auth. When signing in with a magic link login, I'm receiving the auth email to Inbucket. The link in the email to continue the auth process is missing the port though:
`https://localhost/auth/...`
If I manually append the port and load the page it works fine:
`https://localhost:54321/auth/...`

Anyone run into this? I'm not sure if this is an error in my config, or a bug somewhere along the way. I also haven't been able to work out how to configure the GoTrue emails in the local development case to be able to try and unpack it further.

[config.toml.txt](https://github.com/supabase/supabase/files/12114155/config.toml.txt)

**Comment by silentworks** [comment]
Yeah I noticed this in my email_change email too, but not in magic links. What version of the CLI are you using?

**Comment by tomelliot** [comment]
```
$ npx supabase --version
1.77.9
```

**Comment by coryvirok** [comment]
This just started happening to me as well. Yesterday everything was working and today is stopped. I'm wondering if it could be that my local environment pulled in a brand new Docker image that caused it to break overnight. 

```
❯ npx supabase --version
1.77.9
```

Note: this is the same supabase version I had yesterday when things were working.

**Reply by coryvirok** [reply]
Investigating further, I think this started happening when the new supabase/gotrue image (v2.82.4) was pulled in. 

Possibly related
- https://github.com/supabase/cli/pull/1341
- https://github.com/supabase/auth-helpers/issues/569
   I no longer get the PKCE flow... error and I'm able to sign up with email. However the email confirmation link I get is missing the API port (as this bug reports.)
- If I issue a new magic link in my local supabase, the link sent to inbucket is also invalid. I'm pretty sure "supabase_king_v2" should be "localhost:54321":
   ```
   Confirm your email address ( http://supabase_kong_v2/auth/v1/verify?token=d494920863fe7c65cf71843443277f56142010b7b176ef5561c32876&type=signup&redirect_to=http://localhost:5173 )
   ```
   ![Screenshot 2023-07-30 at 5 57 03 PM](https://github.com/supabase/supabase/assets/769323/a2aaeceb-1bb4-4855-852a-2c82a3321b23)

**Comment by tomelliot** [comment, marked as answer]
I just updated some supabase dependencies and noticed this issue is fixed:

```
$ npx supabase --version
1.86.2
```

Another issue that's I'm still seeing is that it doesn't respect the assigned redirect_to URL (defaulting to localhost:3000), even if it's assigned in config.toml under additional_redirect_urls, but that's a separate issue.
